### ISSUE TITLE
Another question about the difference between paper and the result i got

### ISSUE THREAD

**Issue by Jerrrrry-Zhu** [NVlabs/DREAM#20]
hello,I have another question and look forward to your generous answer.
When I run the pre-trained model to inference ”azure“ dataset ,I got the right answer as the paper
```
pck_auc = 0.74406,add_auc = 0.69802
```
![1661436547143](https://user-images.githubusercontent.com/61321296/186687622-1dc72fd4-5f1d-4231-91c6-6ae63ea2b284.png)
![1661436617318](https://user-images.githubusercontent.com/61321296/186687903-f834daf1-2126-4c95-a83f-d3295dc7b6bb.png)
But when i run the same model to inference "test dr" dataset,the pck_auc is normal,but as for the add_auc,I got some problem,why my add_auc is very low?
```
pck_auc = 0.84631,add_auc = 0.00021
```
![1661436711823](https://user-images.githubusercontent.com/61321296/186688274-177ec63e-f53e-42a2-a5d3-450c2205dc63.png)
![1661436778844](https://user-images.githubusercontent.com/61321296/186688561-12c4d38f-4603-40fa-b5b1-9ff3b46469e5.png)

**Comment by tabula-rosa** [comment]
Hello, and thanks for your continued interest in DREAM!

This issue arises because the synthetic datasets have different units -- their data is stored in cm, whereas the real camera datasets use m.

To fix this, replace [this line](https://github.com/NVlabs/DREAM/blob/d71b1760b02f831f613229838e0ae9ea39340100/dream/analysis.py#L355):
```
pnp_results = pnp_metrics(pnp_add, all_n_inframe_projs_gt)
```

with this code:
```
pnp_add_divided = np.array(pnp_add)/100.
pnp_results = pnp_metrics(pnp_add_divided, all_n_inframe_projs_gt, pnp_magic_number=-9.99)
```

Then, you can run this on one of the synthetic datasets:
```
python scripts/network_inference_dataset.py -i trained_models/panda_dream_vgg_q.pth -d data/synthetic/panda_synth_test_dr/ -o temp/panda_dream_vgg_q_synth_panda_test_dr_snc -p shrink-and-crop
```

In this example, I obtain a PCK AUC of 0.77669 and an ADD AUC of 0.78796, both of which match the paper.

I hope that helps!

**Comment by Jerrrrry-Zhu** [comment]
Thank you for your answer,and i am also confused about the [add_from_pose funtion](https://github.com/NVlabs/DREAM/blob/d71b1760b02f831f613229838e0ae9ea39340100/dream/geometric_vision.py#L183) .
![1662009161993](https://user-images.githubusercontent.com/61321296/187836326-4d3ee547-fe06-44db-995a-5667dbd99de8.png)
the **kp_pos_gt_homog** is stack from the **keypoint_positions_wrt_cam_gt**.

the **transform matrix** is solved by the pnp,which can transform the **keypoint_positions_wrt_world** to the **keypoint_positions_wrt_camera**.

why  **kp_pos_aligned** is the result of **transform matrix** multiply the **keypoint_positions_wrt_cam** rather than the **keypoint_positions_wrt_world**?

Addtionaly,if the **kp_pos_aligned**  is the reuslt of **transform matrix** multiply the **keypoint_positions_wrt_cam_gt**,when it comes to calculate the error,why is  **kp_pos_aligned** - **keypoint_positions_wrt_cam_gt** ？In theory, only if my **transformation matrix** is an **eye matrix**, the error calculated in this way will be 0.

Look forward to your generous answer!

**Comment by tabula-rosa** [comment]
This is because the variables `translation` and `quaternion` are from the output of `solve_pnp` (for example, [here](https://github.com/NVlabs/DREAM/blob/d71b1760b02f831f613229838e0ae9ea39340100/dream/analysis.py#L326)). The way the code is structured is that if ground truth values are provided to `solve_pnp`, then `translation` and `quaternion` would represent the identity transform. This is because the output of `solve_pnp` is the transformation that aligns these reference frames --- it returns the pose _difference_ with respect to the camera. The variable names `translation` and `quaternion` in `add_from_pose` are perhaps slightly misleading, and maybe best would be named `delta_translation` and `delta_quaternion`.

Additionally, the discussion in Issue #11 may provide more insight as to our implementation of the ADD metric.

I hope that helps!

**Comment by Jerrrrry-Zhu** [comment]
Got it,that is really an excellent operation,Thank for your reply!
